python/ray/autoscaler/_private/local/config.py

```diff
@@ -1,2 +1,94 @@
-def bootstrap_local(config):
+import copy
+from typing import Any
+from typing import Dict
+
+from ray.autoscaler._private.cli_logger import cli_logger
+
+unsupported_field_message = ("The field {} is not supported "
+                             "for on-premise clusters.")
+
+
+def prepare_local(config: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Prepare local cluster config for ingestion by cluster launcher and
+    autoscaler.
+    """
+    config = copy.deepcopy(config)
+    for field in "head_node", "worker_nodes", "available_node_types":
+        if config.get("head_node"):
+            err_msg = unsupported_field_message.format(field)
+            cli_logger.error(err_msg)
+    if "coordinator_address" in config["provider"]:
+        config = prepare_coordinator(config)
+    else:
+        config = prepare_manual(config)
+    # Presence of head_node and worker_nodes fields triggers legacy config
+    # processing later in the bootstrap process.
+    config["head_node"] = {"placeholder": {}}
+    config["worker_nodes"] = {"placeholder": {}}
+    return config
+
+
+def prepare_coordinator(config: Dict[str, Any]) -> Dict[str, Any]:
+    config = copy.deepcopy(config)
+    # User should explicitly set the max number of workers for the coordinator
+    # to allocate.
+    if "max_workers" not in config:
+        cli_logger.error("The field `max_workers` is required when using an "
+                         "automatically managed on-premise cluster.")
+    config.setdefault("min_workers", 0)
+    return config
+
+
+def prepare_manual(config: Dict[str, Any]) -> Dict[str, Any]:
+    config = copy.deepcopy(config)
+    if "worker_ips" not in config["provider"]:
+        cli_logger.error("Please supply a list of `worker_ips` "
```

Comment:
Could we add a quick test for this?

Reply:
Added to testValidateDocker in test_autoscaler_yaml.py